core/app/models/spree/promotion_duplicator.rb

```diff
@@ -0,0 +1,30 @@
+class Spree::PromotionDuplicator
```

Comment:
The methods in this class seems like it should be in `Promotion`. Then we wouldn't need a separate duplicator.

What do you think about having `promotion.duplicate` and the logic of how to duplicate rules and actions in promotion?